addons/website_sms/models/website_visitor.py

```diff
@@ -8,35 +8,35 @@
 class WebsiteVisitor(models.Model):
     _inherit = 'website.visitor'
 
-    def _prepare_visitor_send_sms_values(self):
-        if self.partner_id.mobile:
-            return {
-                'res_model': 'res.partner',
-                'res_id': self.partner_id.id,
-                'partner_ids': [self.partner_id.id],
-                'number_field_name': 'mobile',
-            }
-        return {}
+    def _check_for_sms_composer(self):
+        """ Purpose of this method is to actualize visitor model prior to contacting
+        him. Used notably for inheritance purpose, when dealing with leads that
+        could update the visitor model. """
+        if not self.partner_id or (not self.partner_id.mobile and not self.partner_id.phone):
+            return False
+        return True
```

Comment:
`return self.partner_id and (self.partner_id.mobile or self.partner_id.phone)`

Reply:
Good idea. I updated code abotu mail composer at the same time.